docs/concepts/pull-dependencies.md

```diff
@@ -0,0 +1,202 @@
+# Pull Dependencies
+
+## Overview
+
+The interoperability of canisters on the Internet Computer (IC) is an important feature. 
+
+`dfx` provides a consistent developer workflow for integrating thrid party canisters.
+
+A service provider attaches [necessary metadata](canister-metadata.md) to the canister wasm.
+
+A service consumer then can pull dependencies directly from the IC mainnet and easily deploy them on a local replica.
+
+This document describes the consumer workflow and explains what happens behind the scene.
+
+## Workflow
+
+### 1. Declare "pull" dependencies in `dfx.json`
+
+Below is an example `dfx.json`, the service consumer is developing the "app" canister which has two pull dependencies:
+
+- "dep_b" which has canister ID of "yhgn4-myaaa-aaaaa-aabta-cai" on the mainnet.
+- "dep_c" which has canister ID of "yahli-baaaa-aaaaa-aabtq-cai" on the mainnet.
+
+```json
+{
+    "canisters": {
+        "app": {
+            "type": "motoko",
+            "main": "src/main.mo",
+            "dependencies": [
+                "dep_b", "dep_c"
+            ]
+        },
+        "dep_b": {
+            "type": "pull",
+            "id": "yhgn4-myaaa-aaaaa-aabta-cai"
+        },
+        "dep_c": {
+            "type": "pull",
+            "id": "yahli-baaaa-aaaaa-aabtq-cai"
+        }
+    }
+}
+```
+
+### 2. Pull the dependencies using `dfx deps pull`
+
+Running `dfx deps pull --network ic` will:
+
+1. resolve the dependency graph by fetching `dfx:deps` metadata recuresively;
+2. download wasm of all direct and indirect dependencies from `dfx:wasm_url` into shared cache;
+3. hash check the downloaded wasm against `dfx:wasm_hash` metadata or the hash of the mainnet running canister;
+4. extract `candid:args`, `candid:service`, `dfx:init` from the downloaded wasm;
+5. create `deps/` folder in project root;
+6. save `candid:service` of direct dependencies as `deps/<NAME>.did`;
+7. save `deps/pulled.json` which contains major info of all direct and indirect dependencies;
+
+For the example project, you will find following files in `deps/`:
+
+- `dep_b.did` and `dep_c.did`: candid files that can be imported by "app";
+- `pulled.json` which has follwing content:
+
+```
+{
+  "canisters": {
+    "yofga-2qaaa-aaaaa-aabsq-cai": {
+      "name": null,
```

Comment:
Please make it so the serializer omits this field if None, rather than writing it as `null` in the JSON.